Runtime 6.2.0 and 5.2.0 release notes (#1805)
astronomer/docs: software_versioned_docs/version-0.30/runtime-release-notes.md

- Upgraded `astronomer-providers` to 1.11.2, which includes a collection of bug fixes. See the [`astronomer-providers` changelog](https://github.com/astronomer/astronomer-providers/blob/main/CHANGELOG.rst#1112-2022-11-19). 
- Upgraded `openlineage-airflow` to 0.17.0, which includes improvements to the OpenLineage spark integration and additional facets for the OpenLineage Python client. See the [OpenLineage changelog](https://github.com/OpenLineage/OpenLineage/releases/tag/0.17.0) for more information.  

## Astro Runtime 6.2.0

- Release date: January 26, 2023

> - Airflow version: 2.4.3

Reviewer: ```suggestion
- Apache Airflow version: 2.4.3
```

> ### Early access Airflow bug fixes

Reviewer: ```suggestion
### Early access Apache Airflow bug fixes
```

> In anticipation of future support for the Kubernetes executor on Astro, Astro Runtime includes the following bug fixes from Airflow 2.5.2:

Reviewer: ```suggestion
In anticipation of future support for the Kubernetes executor on Astro, Astro Runtime includes the following bug fixes from Apache Airflow 2.5.2:
```

- Fix bad pods pickled in executor_config ([28454](https://github.com/apache/airflow/pull/28454))
- Be more selective when adopting pods with KubernetesExecutor ([28899](https://github.com/apache/airflow/pull/28899))

> - Only patch single label when adopting pod ([28776](https://github.com/apache/airflow/pull/28776))

Reviewer: ```suggestion
- Only patch single label when adopting pod ([28776](https://github.com/apache/airflow/pull/28776))
- Don’t re-patch pods that are already controlled by current worker ([26778](https://github.com/apache/airflow/pull/26778))
- Fix backfill queued task getting reset to scheduled state ([23720](https://github.com/apache/airflow/pull/23720))
```

> Astro Runtime 6.0.3 includes same-day support for Airflow 2.4.2. Some changes in Airflow 2.4.2 include:

Reviewer: ```suggestion
Astro Runtime 6.0.3 includes same-day support for Apache Airflow 2.4.2. Some changes in Airflow 2.4.2 include:
```

- Release date: September 30, 2022

> - Airflow version: 2.4.1

Reviewer: ```suggestion
- Apache Airflow version: 2.4.1
```

> ### Airflow 2.4.1

Reviewer: ```suggestion
### Apache Airflow 2.4.1
```

> Astro Runtime 6.0.2 includes same-day support for Airflow 2.4.1, which includes a collection of bug fixes. Fixes include:

Reviewer: ```suggestion
Astro Runtime 6.0.2 includes same-day support for Apache Airflow 2.4.1, which includes a collection of bug fixes. Fixes include:
```

- Fix Deferrable stuck as scheduled during backfill ([#26205](https://github.com/apache/airflow/pull/26205))
- Don't update backfill run from the scheduler ([#26342](https://github.com/apache/airflow/pull/26342))

For a complete list of commits, see the [Apache Airflow 2.4.1 release notes](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#airflow-2-4-1-2022-09-30).

> ### Early access Airflow bug fixes

Reviewer: ```suggestion
### Early access Apache Airflow bug fixes
```

- Backported a fix to correct an issue where logs were not loading from Celery workers ([#26493](https://github.com/apache/airflow/pull/26493))
- Fixed [CVE-2022-40674](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-40674)

## Astro Runtime 6.0.0

- Release date: September 19, 2022

> - Airflow version: 2.4.0

Reviewer: ```suggestion
- Apache Airflow version: 2.4.0
```

> ### Airflow 2.4 and data-aware scheduling

Reviewer: ```suggestion
### Apache Airflow 2.4 and data-aware scheduling
```

> Astro Runtime 6.0.0 provides same-day support for [Airflow 2.4.0](https://airflow.apache.org/blog/airflow-2.4.0/), which delivers significant new features for DAG scheduling. The most notable new features in Airflow 2.4.0 are:

Reviewer: ```suggestion
Astro Runtime 6.0.0 provides same-day support for [Airflow 2.4.0](https://airflow.apache.org/blog/airflow-2.4.0/), which delivers significant new features for DAG scheduling. The most notable new features in Apache Airflow 2.4.0 are:
```

- [Data-aware scheduling](https://airflow.apache.org/docs/apache-airflow/2.4.0/concepts/datasets.html), which is a new method for scheduling a DAG based on when an upstream DAG modifies a specific dataset.

> - The [ExternalPythonOperator](https://airflow.apache.org/docs/apache-airflow/2.4.0/howto/operator/python.html#externalpythonoperator), which can execute Python code in a virtual environment with different Python libraries and dependencies than your core Airflow environment.

Reviewer: ```suggestion
- The [ExternalPythonOperator](https://airflow.apache.org/docs/apache-airflow/2.4.0/howto/operator/python.html#externalpythonoperator), which can execute Python code in a virtual environment with different Python libraries and dependencies than your core Apache Airflow environment.
```

- Automatic DAG registration. You no longer need to specify `as dag` when defining a DAG object.
- Support for [zipping](https://airflow.apache.org/docs/apache-airflow/2.4.0/concepts/dynamic-task-mapping.html#combining-upstream-data-aka-zipping) dynamically mapped tasks.

For a complete list of commits, see the [Apache Airflow 2.4.0 release notes](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#airflow-2-4-0-2022-09-19).

### Additional improvements

- Upgraded `astronomer-providers` to 1.9.0, which includes two new deferrable versions of the operators from the dbt provider package. See the [Astronomer Providers changelog](https://github.com/astronomer/astronomer-providers/blob/1.9.0/CHANGELOG.rst).
- Upgraded `openlineage-airflow` to version `0.14.1`. See the [OpenLineage changelog](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md).

## Astro Runtime 5.2.0

- Release date: January 26, 2023

> - Airflow version: 2.3.4

Reviewer: ```suggestion
- Apache Airflow version: 2.3.4
```

> ### Early access Airflow bug fixes

Reviewer: ```suggestion
### Early access Apache Airflow bug fixes
```

> In anticipation of future support for the Kubernetes executor on Astro, Astro Runtime includes the following bug fixes from Airflow 2.5.2:

Reviewer: ```suggestion
In anticipation of future support for the Kubernetes executor on Astro, Astro Runtime includes the following bug fixes from Apache Airflow 2.5.2:
```

## Astro Runtime 5.0.11

- Release date: November 2, 2022

> - Airflow version: 2.3.4

Reviewer: ```suggestion
- Apache Airflow version: 2.3.4
```

### Additional improvements

> - Upgraded `astronomer-providers` to 1.10.0, which includes two new deferrable versions of operators, `SFTPSensorAsync` and `ExternalDeploymentTaskSensorAsync`. See the [Astronomer Providers changelog](https://github.com/astronomer/astronomer-providers/blob/1.10.0/CHANGELOG.rst).

Reviewer: ```suggestion
- Upgraded `astronomer-providers` to 1.10.0, to include the new  `SFTPSensorAsync` and `ExternalDeploymentTaskSensorAsync` deferrable operators,. See the [Astronomer Providers changelog](https://github.com/astronomer/astronomer-providers/blob/1.10.0/CHANGELOG.rst).
```

- Upgraded `openlineage-airflow` to version `0.15.1`. See the [OpenLineage changelog](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md).

### Bug fixes

- Revert “Cache the custom secrets backend so the same instance gets re-used” ([#25556](https://github.com/apache/airflow/pull/25556))

> - Fixed faulty Kubernetes executor config serialization logic

Reviewer: ```suggestion
- Fixed faulty Kubernetes executor configuration serialization logic
```

## Astro Runtime 5.0.9

- Release date: September 20, 2022
- Airflow version: 2.3.4

> ### Early access Airflow bug fixes

Reviewer: ```suggestion
### Early access Apache Airflow bug fixes
```
